### PR TITLE
Use $(hostname -f) to determine hostname in mod_ssl password script

### PR DESCRIPTION
**Pull request by rcritten**
https://pagure.io/freeipa/issue/7528

Signed-off-by: Rob Crittenden <rcritten@redhat.com>

### CODE REVIEW

**Comment by abbra** [comment]
I had to do a bit of investigation of how actually mod_ssl is running this helper. For `exec:` type of filter it ends up using `fork()` + `execve()` with an empty environment. So this explains why `HOSTNAME` is undefined when the script is run by Apache.

I think using `hostname -f` is sensible here to solve the FQDN problem. This approach assumes our primary hostname is the one vhost is configured to use, though. If you'd have a system with multiple vhosts using the same certificate with SANs, it would not work, though.

May be we should not do this `HOSTNAME` comparison at all? Instead, resolve `${1/:/-}-$2` via `realpath -e --relative-to=/var/lib/ipa/passwds`,  check that it has no `..` or `/` in the result, do a stat on a file `/var/lib/ipa/passwds/$realpath` and if it exists and readable, return its content.

**Comment by rcritten** [comment]
$HOSTNAME should be set by bash AFAICT. It definitely isn't empty since the password script works 99% of the time.

The current script is typical of what an Apache password reader looks like. It may be that Standa left it this way in order to allow for customization.

I'll look into your suggestion, it is an interesting idea. I think I'd prefer to except .. and / from $1 instead as these may be valid characters in a password. This script will run as root so we definitely need to be careful about what it is allowed to do.

**Comment by rcritten** [comment]
I took a slightly different tact. I compare the calculated pwdfile path to that of the output of realpath. In any normal case this should match.

Theoretically this could blow up if some user has a symlink to /var but really...

Note that the way we generate the filename it is going to have a dash in it which will prevent most tricking of filenames anyway.

/usr/libexec/ipa/ipa-httpd-pwdreader ../../../../etc/shadow ''
resolves to
/etc/shadow-

**Comment by abbra** [comment]
@adelton, can you help us here: regarding 'symlink to /var' above -- do you think this could cause issues to freeipa-container?

File: install/tools/ipa-httpd-pwdreader [hunk 1]
```diff
@@ -4,33 +4,28 @@
 # If you'd like to write your custom binary providing passwords to mod_ssl,
 # see the documentation of the aforementioned directive of the mod_ssl module.
 
+hostname=$(hostname -f)
```
Comment (abbra) [inline install/tools/ipa-httpd-pwdreader]: `hostname` variable is not used anymore

**Comment by adelton** [comment]
If the symlinks don't use `..`s and are mostly absolute, things should work fine for freeipa-container.

**Comment by abbra** [comment]
ACK. Rob, I think you want to rebase to get Travis CI status cleared. It is all green but somehow github shows Travis CI run is 'pending'.

**Comment by flo-renaud** [comment]
I relaunched the travis-CI tests and the status got properly updated. Removing the needs_rebase label as it is not required.

**Comment by flo-renaud** [comment]
@rcritten not sure what happened but your commit includes po files, can you fix it?

**Comment by rcritten** [comment]
Gah! Thanks for noticing, I have no idea how those got in there!

**Comment by flo-renaud** [comment]
fedora-28/test_service_permissions and fedora-28/test_topologies failed, adding re-run label

**Comment by rcritten** [comment]
master:

* e3820682c797c8dca86767dc347290f54f872a7c Try to resolve the name passed into the password reader to a file

**Pull request closed without merging**